pkg/test/ginkgo/cmd_runsuite.go

```diff
@@ -236,6 +238,23 @@ func (opt *Options) Run(args []string) error {
 	// anomalies
 	var syntheticTestResults []*JUnitTestCase
 	if events := m.Events(time.Time{}, time.Time{}); len(events) > 0 {
+		// Serialize the interval data for easier external analysis
+		intervalsFilename := filepath.Join(opt.JUnitDir, "intervals.json.gz")
+		if file, err := os.Create(intervalsFilename); err != nil {
+			fmt.Fprintf(opt.Out, "ERROR: failed to create %s: %v\n", intervalsFilename, err)
+		} else {
+			gz := gzip.NewWriter(file)
+			defer func() {
```

Comment:
This defer is going to be super late, you shouldn't use defer here but just have two separate close paths (one for error paths, one for success).

Note you have to close the file anyway to ensure it's flushed.

Reply:
Good catch, fixed (using a single close now because I switched to os.OpenFile)